crates/jstz_proto/src/context/account.rs

```diff
@@ -33,11 +34,37 @@ impl ToString for Nonce {
     }
 }
 
+/// Invariant: if code is present it parses successfully
+#[derive(Default, PartialEq, Eq, Debug, Clone, Serialize, Deserialize)]
+pub struct ParsedCode(String);
+impl From<ParsedCode> for String {
+    fn from(ParsedCode(code): ParsedCode) -> Self {
+        code
+    }
+}
+impl Display for ParsedCode {
+    fn fmt(
+        &self,
+        formatter: &mut std::fmt::Formatter<'_>,
+    ) -> std::result::Result<(), std::fmt::Error> {
+        Display::fmt(&self.0, formatter)
+    }
+}
```

Comment:
nit: usually we avoid `std::...` and prefer `fmt::Formatter` and `fmt::Result<()>`